app/controllers/exam_templates_controller.rb

```diff
@@ -54,10 +54,30 @@ def update
       else
         old_template_filename = old_exam_template.filename
         old_exam_template.replace_with_file(new_uploaded_io.read, assignment_id: assignment.id, filename: old_template_filename)
+        old_exam_template.update(exam_template_params)
         respond_with(old_exam_template, location: assignment_exam_templates_url)
         return
       end
+    else
+      # updating template division
+      if old_exam_template.update(exam_template_params)
+        flash_message(:success, t('exam_templates.update.success'))
+      else
+        flash_message(:error, t('exam_templates.update.failure'))
+      end
     end
     redirect_to action: 'index'
   end
+
+  def exam_template_params
+    params.require(:exam_template)
+      .permit(
+        :assignment,
```

Comment:
When we don't include :id Inside template_divisions_attributes, it doesn't seem to update itself. So I left :id in template_divisions_attributes along with :start, :end, and :label